packages/example/src/pages/index.tsx

```diff
@@ -34,6 +36,17 @@ export default function Index() {
           type="checkbox"
         />
       </label>
+
+      <hr/>
+      <p> after clicking on: <a href="#sync">a hash link</a> the url should contain the hash despite changing params</p>
```

Comment:
Does this work for you? For me the hash is removed when query params change.

![Kapture 2021-11-30 at 08 07 00](https://user-images.githubusercontent.com/4038316/144001539-b313d55a-8621-4494-9af6-c746fcc3a5c1.gif)